### ISSUE TITLE
update httpoison

### ISSUE THREAD

**Issue by EBKH** [edgurgel/tentacat#152]
Is there a reason why not jump to httpoison 1.x?
If not, may I crete the pul request?

**Comment by edgurgel** [comment]
That would be great! Thanks, @EBKH 🎉

**Comment by EBKH** [comment]
@edgurgel I wasn't able to update it, the test found 188 failures, all of them with `(HTTPoison.Error) :req_not_found`
I tried to follow up the functions and end up at `lib/tentacat.ex:96`, but unable to find what went wrong

**Comment by atomicnumber1** [comment]
Hi,
So I was trying the upgrade `HTTPoison` to the latest version `1.4`
Thing is, the `HTTPoison.Base` has `response` type defined in the `__using__` macro which conflicts with the `Tentacat`'s the `response` definition
```
  @type response ::
          {:ok, :jsx.json_term(), HTTPoison.Response.t()} | {integer, any, HTTPoison.Response.t()}
```

How do you think should we approach this problem?

**Comment by edgurgel** [comment]
Good question. Other than renaming the tentacat one, maybe two possibles approaches:

* Remove `response` type from HTTPoison and just use it as `HTTPoison.Response.t()` 
* Move it to `HTTPoison` and out of `HTTPoison.Base` 

What do you think?

**Comment by atomicnumber1** [comment]
Hey
- um... The `__using__` macro is of an external module. How do you propose we remove the type definition from that? I mean we're just consuming the module at this point. We'll need to discuss this with the `HTTPoison` devs if we want to change that module's behaviour, right?
- Same as above, They're both part of the external module.

I was thinking there are two ways to approach the problem,
- Propose a change to the external module `HTTPoison` like you suggested. This will be difficult as it might break other libraries/applications using `HTTPoison` and surely `HTTPoison` devs don't want that.
- Change our internal module.


Now for the internal module change, I was thinking,
- We can say rename the tentacat response as you suggested (ugly)
- We can create an actual wrapping module without doing `use HTTPoison.Base` 

What do you think?

**Comment by edgurgel** [comment]
If we decide to change HTTPoison I think it's fine as we can consider this a bug as we don't *need* to expose the response type as part of the `using` call. 

We could also rename Tentacat's type.

I don't have a strong opinion TBH. What do you think? :D

**Comment by atomicnumber1** [comment]
In that case, we can just make a change to HTTPoison :)

**Comment by sheharyarn** [comment]
Any update on this? Getting a function clause error when attempting to use it with HTTPoison 1.5.x:

```
23:30:04.183 pid=<0.3365.0>  [error] ** (FunctionClauseError) no function clause matching in Tentacat.process_response/1
    (tentacat) lib/tentacat.ex:19: Tentacat.process_response({200, ... ... })
﻿    (tentacat) lib/tentacat.ex:158: Tentacat.request_with_pagination/4
    (tentacat) lib/tentacat.ex:113: Tentacat.request_stream/5

...
```

**Comment by sheharyarn** [comment]
I believe we can close this now that #181 is merged and the `v1.6.x` releases of tentacat are available on hex.

**Comment by edgurgel** [comment]
@sheharyarn, thanks! you are right
